### PR TITLE
Wrap exceptions to JetException is any exception in chain is a HazelcastSerializationException

### PR DESCRIPTION
**Pull request by TomaszGaweda**
If processor throw JetException with a non-serializable cause, then the result cannot be send - `peel` method doesn't see HazecastSerializationException due to JetException not being included in the definition. That's why it's better to just check whole `cause` chain.

See https://github.com/hazelcast/hazelcast/pull/22497/files#r996878225

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java [hunk 1]
```diff
@@ -96,23 +82,24 @@ public static boolean isTopologyException(Throwable t) {
      * Called during startup to make our exceptions known to Hazelcast serialization
      */
     public static void registerJetExceptions(@Nonnull ClientExceptionFactory factory) {
-        for (Tuple3<Integer, Class<? extends Throwable>, ExceptionFactory> exception : EXCEPTIONS) {
-            factory.register(exception.f0(), exception.f1(), exception.f2());
-        }
+        factory.register(JET_EXCEPTIONS_RANGE_START, JetException.class, JetException::new);
```
Comment (TomaszGaweda) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java:85]: 6 lines vs 11 with same readability and no warnings produced by compiler... I think it should be ok to include simple boyscout, WDYT?

**Comment by TomaszGaweda** [comment]
run-nightly-tests

File: hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java [hunk 2]
```diff
@@ -96,23 +82,25 @@ public static boolean isTopologyException(Throwable t) {
...
-        for (Tuple3<Integer, Class<? extends Throwable>, ExceptionFactory> exception : EXCEPTIONS) {
-            factory.register(exception.f0(), exception.f1(), exception.f2());
-        }
+        int id = JET_EXCEPTIONS_RANGE_START;
+        factory.register(id++, JetException.class, JetException::new);
+        factory.register(id++, TopologyChangedException.class, TopologyChangedException::new);
+        factory.register(id++, JobNotFoundException.class, JobNotFoundException::new);
+        factory.register(id++, JobAlreadyExistsException.class, JobAlreadyExistsException::new);
+        factory.register(id++, AssertionCompletedException.class, AssertionCompletedException::new);
+        factory.register(id, JetDisabledException.class, JetDisabledException::new);
```
Comment (ramizdundar) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java:91]: nit:
```suggestion
        factory.register(id++, JetDisabledException.class, JetDisabledException::new);
```

**Review by ramizdundar — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java (hunk 2 above)
Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java:91]: ah, unlucky, ok

File: hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java [hunk 3]
```diff
@@ -92,7 +93,7 @@ private void doSendResponse(Object value) {
                 sendResponse(value);
             } catch (Exception e) {
                 Throwable ex = peel(e);
-                if (value instanceof Throwable && ex instanceof HazelcastSerializationException) {
+                if (value instanceof Throwable && isOrHasCause(ex, HazelcastSerializationException.class)) {
```
Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AsyncOperation.java:96]: Could you please explain how we get a nested `HazelcastSerializationException`. Note that this exception is from `sendResponse`.

**Review by frant-hartm — CHANGES_REQUESTED** [review]
I applaud the effort to improve the state, but we need a test :-)

**Review by orcunc — APPROVED** [review]

**Pull request closed without merging**